ddtrace/span.py

```diff
@@ -334,21 +335,26 @@ def _set_str_tag(self, key, value):
         U+FFFD.
         """
         try:
-            self.meta[key] = ensure_text(value, errors="replace")
+            self._meta[key] = ensure_text(value, errors="replace")
         except Exception as e:
             if config._raise:
                 raise e
             log.warning("Failed to set text tag '%s'", key, exc_info=True)
 
     def _remove_tag(self, key):
         # type: (_TagNameType) -> None
-        if key in self.meta:
-            del self.meta[key]
+        if key in self._meta:
+            del self._meta[key]
 
     def get_tag(self, key):
         # type: (_TagNameType) -> Optional[Text]
         """Return the given tag or None if it doesn't exist."""
-        return self.meta.get(key, None)
+        return self._meta.get(key, None)
+
+    def get_tags(self):
+        # type: () -> _MetaDictType
+        """Return all tags."""
+        return self._meta
```

Comment:
Should this return a copy? 🤔 

as of right now you could do:

```python
# previously
span.meta["key"] = 5.0  # not a string

# now
span.get_tags()["key"] = 5.0  # not a string
```

If instead we returned `self._meta.copy()` it means accessing all of the tags is a heavier operation, but keeps the tags immutable behind our API which is what we want (?).

What do others think?

Reply:
https://github.com/DataDog/dd-trace-py/pull/3188#discussion_r793158834

Reply:
I'll copy both meta and metrics dictionaries